spec/services/training_periods/create_spec.rb

```diff
@@ -67,4 +69,34 @@
       }.to raise_error(ArgumentError, /Unsupported period type/)
     end
   end
+
+  describe '.school_led' do
+    let(:period) { FactoryBot.create(:ect_at_school_period) }
+
+    it 'calls new with the school_led arguments' do
+      allow(TrainingPeriods::Create).to receive(:new).with(any_args).and_call_original
```

Comment:
Is it worth being specific about the args in here since we do care about them?

We could use something like: 

```
expect(TrainingPeriods::Create).to receive(:new).with(
  period:,
  started_on:,
  training_programme: 'school_led'
).and_call_original
```

That way the test a bit tighter.

Reply:
It's a bit tighter but I don't think it makes any difference with regards to the assertion.

Ill get rid of the `with(any_args)` as that's not adding anything to it either and just go with this, so it's clear:

```
allow(TrainingPeriods::Create).to receive(:new).and_return(true)
```